File: algorithms/algorithms-cplusplus/leetcode/Question_0409_Longest_Palindrome.cc
#include <string>
#include <vector>

using namespace std;

class Solution {
 public:
  int longestPalindrome(string s) {
    if (s.empty()) {
      return 0;
    }
    vector<int> vec(256, 0);
    for (char c : s) {
      vec[c]++;
    }
    int res = 0;
    for (int i : vec) {
      res += i / 2 * 2;
    }
    return res == s.size() ? res : res + 1;
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0409_Longest_Palindrome.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0409_Longest_Palindrome {
    public int longestPalindrome(String s) {
        if (s == null || s.length() == 0) {
            return 0;
        }
        int[] counts = new int[256];
        for (int i = 0; i < s.length(); i++) {
            counts[s.charAt(i)]++;
        }
        int res = 0;
        for (int i : counts) {
            res += i / 2 * 2;
        }
        return res == s.length() ? res : res + 1;
    }
}
